MariaDB/server
# MDEV-27660 Spider: remove #ifdef SPIDER_HANDLER_START_BULK_INSERT_HAS_FLAGS

## Description
Remove unnecessary #ifdefs and dead code in Spider.
ha_spider::start_bulk_insert without uint flags is not used because
handler::ha_start_bulk_insert calls start_bulk_insert for each storage engine
with flags. Therefore, dead code and #ifdef related to
SPIDER_HANDLER_START_BULK_INSERT_HAS_FLAGS have been removed.

- [x] The Jira issue number for this PR is: MDEV-27660

## How can this PR be tested?
No modification of test cases.
Remove dead code to make sure it passes the existing test cases.


## Basing the PR against the correct MariaDB version
